system/Controller.php

```diff
@@ -211,9 +211,8 @@ public function validate($rules, array $messages = []): bool
 		}
 
 		$success = $this->validator
-			->withRequest($this->request)
 			->setRules($rules, $messages)
-			->run();
+			->run($this->request->getPost());
```

Comment:
Were you running into an issue with the withRequest() method? If not, no need to change it.

Reply:
the run method accepts the $data as a parameter. It was showing the syntactical error if run() is written. Hence changed.

Reply:
This is an interface function prototype
public function run(array $data, string $group = null): bool;
where $data is not optional

Reply:
Then the correct solution here is to make $data nullable in the Validator's run method. And adjust tests accordingly. 

it's always possible for PUT or other HTTP verbs than POST to provide data to the application.

Reply:
As per the documentation, I have made the changes. It is written to validate only POST params.
Please refer the attachment and link

https://codeigniter4.github.io/userguide/incoming/controllers.html#validating-post-data

![controllers codeigniter4 4 0 0 alpha 5 documentation](https://user-images.githubusercontent.com/8375139/53574640-fa7a5d80-3b95-11e9-87e3-1cde560e4bdc.png)

Do you want me to make changes for other HTTP verbs?

Reply:
Yes, that should be updated. The `withRequest` method handles any of the verbs, which is why that was used in the controller. Looks like I forgot to update the docs. 

Also - if you could break this out into it's own PR that would be great. Simpler to deal with things when it's only one change/type of change in a PR.

Reply:
Sure. I will make a separate PR for the same, where run method $data param is nullable.
I will update the doc as well.